controllers/cluster_upgrade.go

```diff
@@ -372,109 +400,193 @@ func getProjectedVolumeConfigurationFromPod(pod corev1.Pod) *corev1.ProjectedVol
 	return nil
 }
 
-// isPodNeedingUpgradedImage checks whether an image in a pod has to be changed
-func isPodNeedingUpgradedImage(
+func checkPodImageIsOutdated(
+	status postgres.PostgresqlStatus,
 	cluster *apiv1.Cluster,
-	pod corev1.Pod,
-) (oldImage string, targetImage string, err error) {
+) (rollout, error) {
 	targetImageName := cluster.GetImageName()
 
-	pgCurrentImageName, err := specs.GetPostgresImageName(pod)
+	pgCurrentImageName, err := specs.GetPostgresImageName(status.Pod)
 	if err != nil {
-		return "", "", err
+		return rollout{}, err
 	}
 
 	if pgCurrentImageName != targetImageName {
+		canUpgradeImage, err := postgres.CanUpgrade(pgCurrentImageName, targetImageName)
```

Comment:
NOTE: this fixes a bug. Previously the check to verify if the upgrade is possible was checked only when the current image equals the target image